### PR TITLE
🔎 #2269 Search Clinical Data by Donor

### PR DESCRIPTION
**Pull request by demariadaniel**
**Description of changes**

**Type of Change**

- [ ] Bug
- [x] New Feature

### CODE REVIEW

File: src/services/clinical/index.js [hunk 1]
```diff
@@ -162,6 +162,21 @@ const getClinicalData = async (variables, Authorization) => {
   return response;
 };
 
+const getClinicalSearchResults = async (variables, Authorization) => {
+  const { programShortName, filters } = variables;
+
+  const query = new URLSearchParams(filters).toString();
+
+  const url = `${CLINICAL_SERVICE_ROOT}/clinical/program/${programShortName}/clinical-search-results?${query}`;
```
Comment (samrichca) [inline src/services/clinical/index.js:170]: would UrlJoin work here?

Comment (demariadaniel) [reply src/services/clinical/index.js:170]: It does work would you like it added here + on Clinical Search Results / Clinical Error endpoints?

Comment (samrichca) [reply src/services/clinical/index.js:170]: yes please, we run into problems with slashes in URLs sometimes 👍

Comment (demariadaniel) [reply src/services/clinical/index.js]: Updated

File: src/services/clinical/index.js [hunk 2]
```diff
@@ -163,7 +186,10 @@ const getClinicalData = async (variables, Authorization) => {
 };
 
 const getClinicalErrors = async (programShortName, donorIds, Authorization) => {
-  const url = `${CLINICAL_SERVICE_ROOT}/clinical/program/${programShortName}/clinical-errors?donorIds=${donorIds}`;
+  const url = urlJoin(
+    `${CLINICAL_SERVICE_ROOT}/clinical/program/${programShortName}`,
```
Comment (samrichca) [inline src/services/clinical/index.js:190]: is it possible to separate the env variable from your string in the URL join arguments?

urlJoin(CLINICAL_SERVICE_ROOT, '/clinical...

File: src/routes/clinical-proxy.ts [hunk 3]
```diff
@@ -102,4 +102,21 @@ router.use(
...
 );
 
+router.use(
+  '/program/:programId/clinical-search-results',
+  createProxyMiddleware({
+    target: CLINICAL_SERVICE_ROOT,
+    pathRewrite: (pathName: string, req: Request) => {
+      const programId = req.params.programId;
+      return urlJoin(
+        '/clinical/program/',
```
Comment (samrichca) [inline src/routes/clinical-proxy.ts:112]: this one here is a good pattern for using UrlJoin - separate variables from strings

File: src/services/clinical/index.js [hunk 4]
```diff
@@ -152,7 +153,29 @@ const getClinicalData = async (variables, Authorization) => {
...
+  return response;
+};
+
+const getClinicalSearchResults = async (variables, Authorization) => {
+  const { programShortName, filters } = variables;
+
+  const query = new URLSearchParams(filters).toString();
+
+  const url = urlJoin(
+    `${CLINICAL_SERVICE_ROOT}/clinical/program/${programShortName}`,
```
Comment (samrichca) [inline src/services/clinical/index.js:176]: please separate variables from strings in this url join

Comment (demariadaniel) [reply src/services/clinical/index.js]: Updated

File: src/routes/clinical-proxy.ts (hunk 3 above)
Comment (demariadaniel) [reply src/routes/clinical-proxy.ts:112]: Updated

File: src/services/clinical/index.js (hunk 2 above)
Comment (demariadaniel) [reply src/services/clinical/index.js]: Updated

**Review by samrichca — APPROVED** [review]
using search params and urljoin, great! 👍 👍

**Pull request merged**